frontend/main-app/src/components/Login/Login.tsx

```diff
@@ -1,32 +1,35 @@
 import classes from "./Login.module.css";
 import "@digdir/design-system-tokens/brand/digdir/tokens.css";
-import { Alert, Button, Heading, Textfield } from "@digdir/design-system-react";
+import { Button, Heading, Textfield } from "@digdir/design-system-react";
 import React, { FormEvent, useState } from "react";
-import { useNavigate } from "react-router-dom";
-import { axiosPostForm, getApiUrl, validateLoginForm } from "./LoginUtils";
+import { useNavigate, useParams } from "react-router-dom";
+import { submitForm, getApiUrl, validateLoginForm } from "./LoginUtils";
 import { LoginForm, LoginFormError } from "./types";
 import { useTranslation } from "react-i18next";
+import { alertToRender } from "../FormViewer/validationUtils";
 
 export const Login = (): React.JSX.Element => {
+  const { authError } = useParams();
   const navigate = useNavigate();
-  const [formErrors, setFormErrors] = useState<LoginFormError | null>(null);
   const { t } = useTranslation();
-  const [serverError, setServerError] = useState<string | null>(null);
+
+  const [fieldErrors, setFieldErrors] = useState<LoginFormError | null>(null);
+  const [errorAlert, setErrorAlert] = useState(authError);
 
   const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
     event.preventDefault();
 
     const formData = new FormData(event.currentTarget as HTMLFormElement);
     const loginForm: LoginForm = Object.fromEntries(formData);
-    const formIsValid: boolean = validateLoginForm({ loginForm, setFormErrors });
+    const formIsValid: boolean = validateLoginForm({ loginForm, setFieldErrors });
 
     if (formIsValid) {
       const apiUrl = getApiUrl();
       const targetUrl = `${apiUrl}/api/users/login`;
-      if (await axiosPostForm(targetUrl, formData)) {
+      if (await submitForm(targetUrl, formData)) {
         navigate("/dashboard");
       } else {
-        setServerError(t("signup_page.server.error"));
+        setErrorAlert("loginServerError");
```

Comment:
What if we refactur handleSubmit to :

  ```
const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const formData = new FormData(event.currentTarget);
    const loginForm: LoginForm = Object.fromEntries(formData);
    const formIsValid: boolean = validateLoginForm({ loginForm, setFieldErrors });
    const apiUrl = getApiUrl();
    const targetUrl = `${apiUrl}/api/users/login`;
    try {
      const response = await submitForm(targetUrl, formData);
      if (response && formIsValid) navigate("/dashboard");
    }catch(error){
      setErrorAlert("loginServerError")
    }
  };
```

Reply:
I see that we can reduce nesting, but the API request should not be made if the form is invalid.

What do you think about this?

```
const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    const formData = new FormData(event.currentTarget as HTMLFormElement);
    const loginForm: LoginForm = Object.fromEntries(formData);
    const formIsValid = validateLoginForm({ loginForm, setFieldErrors });

    if (!formIsValid) {
      return;
    }

    const apiUrl = getApiUrl();
    const targetUrl = `${apiUrl}/api/users/login`;

    try {
      await axios.post(targetUrl, loginForm, {
        withCredentials: true,
      });
      navigate("/dashboard");
    } catch (error) {
      console.log(error);
      setErrorAlert("loginServerError");
    }
  };
```

Reply:
Looks good also😊